### PR TITLE
[trcomp] [huggingface_tensorflow] [build] Adding support for TF 2.8

### PR DESCRIPTION
**Pull request by codeislife99**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: huggingface/tensorflow/buildspec-trcomp.yml [hunk 1]
```diff
@@ -24,8 +24,8 @@ images:
     tag_python_version: &TAG_PYTHON_VERSION py38
```
Comment (Lokiiiiii) [inline huggingface/tensorflow/buildspec-trcomp.yml:24]:
```suggestion
    tag_python_version: &TAG_PYTHON_VERSION py39
```

File: huggingface/tensorflow/training/docker/2.8/py3/cu112/Dockerfile.trcomp.gpu [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# https://github.com/aws/deep-learning-containers/blob/master/available_images.md
+# refer to the above page to pull latest Tensorflow image
+
+# docker image region us-west-2
+FROM 763104351884.dkr.ecr.us-west-2.amazonaws.com/tensorflow-training:2.8.0-gpu-py39-cu112-ubuntu20.04-sagemaker
```
Comment (Lokiiiiii) [inline huggingface/tensorflow/training/docker/2.8/py3/cu112/Dockerfile.trcomp.gpu:5]:
```suggestion
FROM 763104351884.dkr.ecr.us-west-2.amazonaws.com/tensorflow-training:2.8-gpu-py39-cu112-ubuntu20.04-sagemaker
```

File: huggingface/tensorflow/training/docker/2.8/py3/cu112/Dockerfile.trcomp.gpu [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+LABEL maintainer="Amazon AI"
+LABEL dlc_major_version="1"
+
+# version args
```
Comment (Lokiiiiii) [inline huggingface/tensorflow/training/docker/2.8/py3/cu112/Dockerfile.trcomp.gpu:10]:
```suggestion
# Version args can be overwritten by buildspec args
```

File: huggingface/tensorflow/training/docker/2.8/py3/cu112/Dockerfile.trcomp.gpu [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+ARG TRANSFORMERS_VERSION=4.18.0
+ARG DATASETS_VERSION=1.18.4
+ARG PYTHON=python3
+
+# Install  Sagemaker PythonSDK
+RUN pip install --no-cache-dir sagemaker
+
+# install Hugging Face libraries and its dependencies
+RUN pip install --no-cache-dir \
+    transformers[sklearn,sentencepiece]==${TRANSFORMERS_VERSION} \
```
Comment (Lokiiiiii) [inline huggingface/tensorflow/training/docker/2.8/py3/cu112/Dockerfile.trcomp.gpu:20]:
```suggestion
    transformers[sklearn,sentencepiece,audio,vision]==${TRANSFORMERS_VERSION} \
```

File: huggingface/tensorflow/training/docker/2.8/py3/cu112/Dockerfile.trcomp.gpu [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    datasets==${DATASETS_VERSION} \
+    tensorflow-addons==0.16.1 \
+    psutil \
+    # TODO: Remove upgrade statements once packages are updated in base image
```
Comment (Lokiiiiii) [inline huggingface/tensorflow/training/docker/2.8/py3/cu112/Dockerfile.trcomp.gpu:24]: These should be fixed by now

File: huggingface/tensorflow/training/docker/2.8/py3/cu112/Dockerfile.trcomp.gpu [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
+RUN apt-get update \
+ # TODO: Remove upgrade statements once packages are updated in base image
+ && apt-get -y upgrade --only-upgrade systemd openssl cryptsetup \
+ && apt install -y git-lfs \
+ && apt-get clean \  
+ && rm -rf /var/lib/apt/lists/*
+
+# Install TensorFlow
+ARG TF_URL="https://framework-binaries.s3.us-west-2.amazonaws.com/tensorflow/hopper/r2.8_aws/gpu/2022-04-20-19-24/tensorflow_gpu-2.8.0-cp39-cp39-manylinux_2_12_x86_64.manylinux2010_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl"
```
Comment (Lokiiiiii) [inline huggingface/tensorflow/training/docker/2.8/py3/cu112/Dockerfile.trcomp.gpu:41]: Remove codeword names

**Review by Lokiiiiii — CHANGES_REQUESTED** [review]

**Comment by tejaschumbalkar** [comment]
TF 2.8 is out of support

**Pull request closed without merging**